src/e2e-test/java/io/cdap/plugin/BQValidation.java

```diff
@@ -0,0 +1,267 @@
+ */
+
+package io.cdap.plugin;
+
+import com.google.cloud.bigquery.BigQuery;
+import com.google.cloud.bigquery.BigQueryOptions;
+import com.google.cloud.bigquery.Field;
+import com.google.cloud.bigquery.FieldValue;
+import com.google.cloud.bigquery.FieldValueList;
+import com.google.cloud.bigquery.Schema;
+import com.google.cloud.bigquery.TableId;
+import com.google.cloud.bigquery.TableResult;
+import com.google.gson.Gson;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import io.cdap.e2e.utils.BigQueryClient;
+import io.cdap.e2e.utils.PluginPropertyUtils;
+import io.cdap.plugin.utils.SalesforceClient;
+import io.cdap.plugin.utils.enums.SObjects;
+import org.junit.Assert;
+
+import java.io.IOException;
+import java.sql.Date;
+import java.time.LocalTime;
+import java.time.OffsetDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+
+/**
+ * BQValidation.
+ */
+public class BQValidation {
+
+  static List<JsonObject> bigQueryResponse = new ArrayList<>();
+  static List<Object> bigQueryRows = new ArrayList<>();
+  static Gson gson = new Gson();
+
+  public static boolean validateSalesforceToBQRecordValues(String objectName, String targetTable) throws
+    IOException, InterruptedException {
+    String uniqueRecordId = SalesforceClient.queryObjectId(objectName);
+
+    List<JsonObject> bigQueryResponse = new ArrayList<>();
+    getBigQueryTableData(targetTable, bigQueryRows);
+    for (Object rows : bigQueryRows) {
+      JsonObject jsonData = gson.fromJson(String.valueOf(rows), JsonObject.class);
+      bigQueryResponse.add(jsonData);
+    }
+    SalesforceClient.queryObject(uniqueRecordId, objectName);
+    return compareSalesforceAndJsonData(SalesforceClient.sobjectResponse, bigQueryResponse, targetTable);
+
+  }
+
+  public static boolean validateBQToSalesforceRecordValues(String targetTable, String objectName) throws
+    IOException, InterruptedException {
+    String uniqueRecordId = SalesforceClient.queryObjectId(objectName);
+
+    getBigQueryTableData(targetTable, bigQueryRows);
+    for (Object rows : bigQueryRows) {
+      JsonObject jsonData = gson.fromJson(String.valueOf(rows), JsonObject.class);
+      bigQueryResponse.add(jsonData);
+    }
+    SalesforceClient.queryObject(uniqueRecordId, objectName);
+    return compareSalesforceAndJsonData(SalesforceClient.sobjectResponse, bigQueryResponse, targetTable);
+
+  }
+
+  public static boolean validateSalesforceMultiObjectToBQRecordValues() throws IOException, InterruptedException {
+    List<String> objectNames = Arrays.asList(SObjects.AUTOMATION_CUSTOM__C.value, SObjects.AUTOMATION_CUSTOM2__C.value);
+    List<String> targetTables = getTableByName();
+    if (objectNames.size() != targetTables.size()) {
+      throw new IllegalArgumentException("Number of object names and target tables must be the same.");
+    }
+
+    for (int i = 0; i < objectNames.size(); i++) {
```

Comment:
please use a different name for the variable, like currentObject etc

Reply:
done

Reply:
nit: instead of using `i` as a variable name, can we use something meaningful like `objNameIndex`

Reply:
done @b164706e5fce61a001b59979c1a2e73e5b70fea7